assets/js/filter.js

```diff
@@ -0,0 +1,196 @@
+
+            if (query.startsWith("<")) {
+                if (query.startsWith("<=")) {
+                    return [this.lteFilter, querySliceTwo];
+                } else {
+                    return [this.ltFilter, querySliceOne];
+                }
+
+            } else if (query.startsWith(">")) {
+                if (query.startsWith(">=")) {
+                    return [this.gteFilter, querySliceTwo];
+                } else {
+                    return [this.gtFilter, querySliceOne];
+                }
+
+            } else if (query.startsWith("==")) {
+                return [this.eFilter, querySliceTwo];
+
+            } else if (query.startsWith("!=")) {
+                return [this.neFilter, querySliceTwo];
+
+            } else if (query.match(/\//g) && query.match(/\//g).length >= 2) {
+                var parts = query.split('/');
+                var regex = query;
+                var options = "";
+
+                if (parts.length > 1) {
+                    regex = parts[1];
+                    options = parts[2];
+                }
+                try {
+                    // If no error is thrown here, the query is valid regex
+                    return [this.regexFilter, new RegExp(regex, options)];
+                } catch (e) {
+                    // If the query is not valid regex, use normal filter
+                    return [this.normalFilter, query];
+                }
+
+            } else {
+                return [this.normalFilter, query];
+            }
+
+        },
+
+        lteFilter: function(text, query) {
+            return text <= query;
+        },
+
+        ltFilter: function(text, query) {
+            return text < query;
+        },
+
+        gteFilter: function(text, query) {
+            return text >= query;
+        },
+
+        gtFilter: function(text, query) {
+            return text > query;
+        },
+
+        eFilter: function(text, query) {
+            return text == query;
+        },
+
+        neFilter: function(text, query) {
+            return text != query;
+        },
+
+        regexFilter: function(text, regExp) {
+            return regExp.test(text);
+        },
+
+        normalFilter: function(text, query) {
+            return text.toLowerCase().indexOf(query.toLowerCase()) >= 0;
+        },
+
+        filterColumn: function(input) {
+            var column = input.attr('data-column');
+            var query = input.val();
+            this.filters[column] = this.filterGetType(query.trim());
+            this.filterTable();
+        },
+
+        filterTable: function() {
+            var self = this;
+            this.element.find('tbody').find('tr').not('.no-filtering').hide().filter(function() {
+                var pass = true;
+                $(this).find('td').each(function(i) {
+                    if (self.filters[i][1] && !self.filters[i][0].apply(this, [$(this).text().trim(), self.filters[i][1]])) {
```

Comment:
Can you break these to vars `const function = self.filters[i][0], argument = self.filters[i][1];`